File: 0238-product-of-array-except-self/0238-product-of-array-except-self.go
func productExceptSelf(nums []int) []int {
    result := make([]int, len(nums))
    prefix := make([]int, len(nums))
    postfix := make([]int, len(nums))

    prefix[0] = 1
    postfix[len(nums) - 1] = 1

    for i := 1; i < len(nums); i++ {
        prefix[i] = nums[i - 1] * prefix[i - 1]
    } 

    for i := len(nums) - 2; i >= 0; i-- {
        postfix[i] = nums[i + 1] * postfix[i + 1] 
    }

    for i := 0; i < len(nums); i++ {
        result[i] = prefix[i] * postfix[i]
    }

    return result
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        prefix = []
        postfix = [-1 for _ in nums]
        prod = 1
        for num in nums:
            prod *= num
            prefix.append(prod)
            
        
        prod = 1
        for i in range(len(nums) - 1, -1, -1):
            prod *= nums[i]
            postfix[i] = prod
        
        result = []
        for i in range(len(nums)):
            if i == 0:
                result.append(postfix[i + 1])
            elif i == len(nums) - 1:
                result.append(prefix[i - 1])
            else:
                result.append(prefix[i - 1] * postfix[i + 1])
        
        return result
